pcsx2-qt/Debugger/BreakpointDialog.cpp

```diff
@@ -140,6 +142,22 @@ void BreakpointDialog::accept()
 
 		mc->start = startAddress;
 		mc->end = startAddress + size;
+		
+		if (!m_ui.txtCondition->text().isEmpty())
+		{
+			mc->hasCond = true;
+			mc->cond.debug = m_cpu;
+
+			if (!m_cpu->initExpression(m_ui.txtCondition->text().toLocal8Bit().constData(), expr))
+			{
+				QMessageBox::warning(this, tr("Error"), tr("Invalid condition \"%1\"").arg(getExpressionError()));
+				return;
+			}
+
+			mc->cond.expression = expr;
+			strncpy(&mc->cond.expressionString[0], m_ui.txtCondition->text().toLocal8Bit().constData(),
```

Comment:
Be mindful that this will not include a null terminator if the source is exactly the length of the buffer. Is there no `strlcpy`-like function in the codebase?